tests/apps/utils/tls.go

```diff
@@ -0,0 +1,115 @@
+/*
+Copyright 2022 The Dapr Authors
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// This utility adapted from https://go.dev/src/crypto/tls/generate_cert.go
+
+package utils
+
+import (
+	"crypto/ecdsa"
+	"crypto/elliptic"
+	"crypto/rand"
+	"crypto/x509"
+	"crypto/x509/pkix"
+	"encoding/pem"
+	"log"
+	"math/big"
+	"net"
+	"os"
+	"path"
+	"strings"
+	"time"
+)
+
+// GenerateTLSCertAndKey generates a self-signed X.509 certificate for a TLS server.
+// Outputs to 'cert.pem' and 'key.pem' and will overwrite existing files.
+//
+// host: Comma-separated hostnames and IPs to generate a certificate for
+// validFrom: The time the certificate is valid from
+// validFor: The duration the certificate is valid for
+// directory: Path to write the files to
+func GenerateTLSCertAndKey(host string, validFrom time.Time, validFor time.Duration, directory string) error {
+	// *********************
+	// Generate private key
+	// *********************
+	tlsKey, err := ecdsa.GenerateKey(elliptic.P256(), rand.Reader)
+	if err != nil {
+		return err
+	}
+
+	b, err := x509.MarshalECPrivateKey(tlsKey)
```

Comment:
More like a nit :) "MarshalECPrivateKey" uses the "legacy" format for encoding the key to ASN.1. It is generally better to use PKCS#8 ([`MarshalPKCS8PrivateKey`](https://pkg.go.dev/crypto/x509#MarshalPKCS8PrivateKey)) and a PEM block of type `PRIVATE KEY` (without `EC`).